lib/Sema/CSDiag.cpp

```diff
@@ -2325,6 +2325,51 @@ bool FailureDiagnosis::diagnoseContextualConversionError(
       diagnose(expr->getLoc(), diag::missing_nullary_call, srcFT->getResult())
         .highlight(expr->getSourceRange())
         .fixItInsertAfter(expr->getEndLoc(), "()");
+
+      // It is possible that we're looking at a stored property being
+      // initialized with a closure. Something like:
+      //
+      // var foo: Int = { return 0 }
+      //
+      // Let's offer another fix-it to remove the '=' to turn the stored
+      // property into a computed property. If the variable is immutable, then
+      // replace the 'let' with a 'var'
+
+      // First, check if the variable is declared top level or not. If it is
+      // not, then it means we're inside a decl like a class or an extension.
+      PatternBindingDecl *PBD = nullptr;
+
+      if (auto TLCD = dyn_cast<TopLevelCodeDecl>(CS.DC)) {
+        if (TLCD->getBody()->isImplicit()) {
+          if (auto decl = TLCD->getBody()->getElement(0).dyn_cast<Decl *>()) {
+            if (auto binding = dyn_cast<PatternBindingDecl>(decl)) {
+              PBD = binding;
+            }
+          }
+        }
+      } else if (auto PBI = dyn_cast<PatternBindingInitializer>(CS.DC)) {
+        PBD = PBI->getBinding();
+      }
+
+      if (PBD) {
+        if (auto VD = PBD->getSingleVar()) {
+          auto entry = PBD->getPatternEntryForVarDecl(VD);
+
+          if (!VD->isStatic() &&
+              !VD->getAttrs().getAttribute<DynamicReplacementAttr>() &&
+              entry.getInit() && isa<ClosureExpr>(entry.getInit())) {
```

Comment:
I don't think this is necessary, but I am not fully sure. I think the check for function type above takes care of this.